### PR TITLE
Re-wrap errors

### PR DESCRIPTION
**Pull request by jsoriano**
Use standard library for errors wrapping.

Most code changes done with `github.com/xdg-go/go-rewrap-errors` and `goimports`.

Steps followed to prepare this PR:
* `find . -name "*.go" -exec go run github.com/xdg-go/go-rewrap-errors@latest -w {} \;`
* `find . -name "*.go" -exec goimports -w {} \;`
* `go mod tidy`
* Manually replaced uses of `log.Fatal(fmt.Errorf` with `log.Fatalf(...` (not really related, but took the opportunity to fix this).
* `make check-static`, and fix linting issues related to error messages.
* Fix case with nil error where `errors.Wrap` would have returned nil (https://github.com/elastic/elastic-package/pull/1332/commits/6993819765ad15966fcc03c796fb37d1aa5a8b32).

### CODE REVIEW

**Review by mrodm — APPROVED** [review]
Nice to have all these occurrences replaced, thanks!

File: internal/packages/changelog/yaml.go [hunk 1]
```diff
@@ -100,19 +103,19 @@ func SetManifestVersion(d []byte, version string) ([]byte, error) {
...
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to decode manifest")
+		return nil, fmt.Errorf("failed to decode manifest: %w", err)
 	}
 
 	// Manifest is a document, with a single element, that should be a map.
 	if len(node.Content) == 0 || node.Content[0].Kind != yaml.MappingNode {
-		return nil, errors.Wrap(err, "unexpected manifest content")
+		return nil, fmt.Errorf("unexpected manifest content: %w", err)
 	}
```
Comment (efd6) [inline internal/packages/changelog/yaml.go:112]: This is not right; err is nill here but this returns a non-nil error referring to err.

Comment (jsoriano) [reply internal/packages/changelog/yaml.go:112]: Thanks, good catch.

**Pull request merged**